### ISSUE TITLE
GPU histc returns tensor of wrong dtype

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#20208]
cc @xvdp @jaciefan 

Original report from @xvdp: https://github.com/pytorch/pytorch/issues/1382#issuecomment-490092747

Thank you for the implementation, it is present in latest release, but there's an inconsistency in behaviour you may want to fix: cuda and cpu return different types

```
import torch
import sys
print(sys.version_info)
print(torch.__version__)
normalcpu = torch.zeros(100, dtype=torch.float32, device="cpu", requires_grad=False).normal_(0., 1.)
h_cpu = torch.histc(normalcpu, bins=10)
normalcuda = normalcpu.to(device="cuda")
h_cuda = torch.histc(normalcuda, bins=10)
print(h_cpu, h_cpu.dtype)
print(h_cuda, h_cuda.dtype)
```

```
sys.version_info(major=3, minor=7, micro=3, releaselevel='final', serial=0)
1.1.0
tensor([ 2.,  5., 15., 26., 17., 18., 10.,  4.,  2.,  1.]) torch.float32
tensor([ 2,  5, 15, 26, 17, 18, 10,  4,  2,  1], device='cuda:0') torch.int64
```

**Comment by zou3519** [comment]
Probably my fault. I'll assign it to myself when I get to working on it

**Comment by gchanan** [comment]
@izdeby can you take a look?

**Comment by yf225** [comment]
Removing "needs discussion" from this issue, as there is already an open and approved PR for it.

**Comment by ezyang** [comment]
Fixed.

**Comment by npv0** [comment]
Does this support include cuda toolkit 10.0.130 & pytorch 1.1.0 
reference: LEDNet  (https://github.com/AceCoooool/LEDNet/issues/8)

**Comment by ezyang** [comment]
No, the fix only made it to 1.2
